src/com/projectkorra/projectkorra/waterbending/healing/HealingWaters.java

```diff
@@ -233,7 +233,7 @@ private void applyHealing(final LivingEntity livingEntity) {
 
 	private boolean inWater(final Player player) {
 		final Block block = player.getLocation().getBlock();
-		return isWater(block) && !TempBlock.isTempBlock(block);
+		return (isWater(block) && !TempBlock.isTempBlock(block)) || (isCauldron(block) && !TempBlock.isTempBlock(block));

```

Comment:
This can be optimized to `(isWater(block) || isCauldron(block)) && !TempBlock.isTempBlock(block)`

Reply:
nice catch👍